### DISCUSSION TITLE
Can I start a new python process for every new user that tries to connect?

### DISCUSSION THREAD

**Discussion by ItsCubeTime** [zauberzeug/nicegui#464] [Q&A] [answered]
Hi 👋 Would it be outrageously stupid to start a new python process for each and every connecting client?
Hoping to avoid issues like: 
A: Some user pressing a button that runs some long running operation on the backend, blocking the main thread, making the entire app unresponsive for a while and it affecting several users
B: Me accidentally storing some temporary data meant to be per-user in the global namespace

What Im most concerned about is probably performance issues, would I be IO bound before my server is likely to get overloaded by all the python processes? Or would I likely be fine if my system has say 128gb of ram or so?

Im also curious if you guys know of any software/libraries that could be used to manage the incoming HTTP requests and send them to the right python process (and also fire up new processes as necessary).

**Comment by rodja** [comment, marked as answer]
Processes are not the best approach. Here some reasons:

- Consumes a lot of system resources, potentially leading to slow response times and system crashes.
Makes it more difficult to share data between clients and maintain state across multiple requests.
- Increases the complexity of your code and makes it harder to maintain.
- May not be the most efficient approach to handling multiple clients.
- Can result in a large number of processes running simultaneously, which can be difficult to manage and monitor.
- May not scale well for larger applications or high volumes of traffic.

This is why FastAPI (which NiceGUI is build upon) uses asyncio with an eventloop. It's much more efficient and counteracts all the above mentioned issues.
